File: medplum-java/medplum-java-model/src/main/java/com/fhir/PaymentReconciliation_Detail.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutablePaymentReconciliation_Detail.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface PaymentReconciliation_Detail {
  Optional<Identifier> predecessor();

  Optional<Identifier> identifier();

  CodeableConcept type();

  Optional<Reference> response();

  Optional<date> date();

  Optional<Money> amount();

  Optional<String> id();

  Optional<Reference> submitter();

  Optional<Reference> payee();

  Optional<List<Extension>> modifierExtension();

  Optional<Reference> responsible();

  Optional<Reference> request();

  Optional<List<Extension>> extension();

  static ImmutablePaymentReconciliation_Detail.TypeBuildStage builder() {
    return ImmutablePaymentReconciliation_Detail.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/PaymentReconciliation_Detail.ts
import { CodeableConcept } from "./CodeableConcept";
import { date } from "./date";
import { Extension } from "./Extension";
import { Identifier } from "./Identifier";
import { Money } from "./Money";
import { Reference } from "./Reference";

export interface PaymentReconciliation_Detail {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier | null | undefined;
    predecessor: Identifier | null | undefined;
    type: CodeableConcept;
    request: Reference | null | undefined;
    submitter: Reference | null | undefined;
    response: Reference | null | undefined;
    date: date | null | undefined;
    responsible: Reference | null | undefined;
    payee: Reference | null | undefined;
    amount: Money | null | undefined;
}
